pkg/streamer/reader.go

```diff
@@ -118,59 +121,21 @@ func (r *BinlogReader) checkRelayPos(pos mysql.Position) error {
 	return nil
 }
 
-// getUUIDByGTID gets uuid subdir which contain the gtid set
-func (r *BinlogReader) getUUIDByGTID(gset mysql.GTIDSet) (string, error) {
-	// get flush logs from oldest to newest
-	for _, uuid := range r.uuids {
-		filename := path.Join(r.cfg.RelayDir, uuid, utils.MetaFilename)
-		var meta Meta
-		_, err := toml.DecodeFile(filename, &meta)
-		if err != nil {
-			return "", terror.ErrRelayLoadMetaData.Delegate(err)
-		}
-
-		gs, err := mysql.ParseGTIDSet(r.cfg.Flavor, meta.BinlogGTID)
-		if err != nil {
-			return "", terror.ErrRelayLoadMetaData.Delegate(err)
-		}
-		if gs.Contain(gset) {
-			r.tctx.L().Info("get uuid subdir by gtid", zap.Stringer("GTID Set", gset), zap.String("uuid", uuid), zap.Stringer("latest GTID Set in subdir", gs))
-			return uuid, nil
-		}
-	}
-
-	// TODO: use a better mechanism to call relay.meta.Flush
-	// get the meta save in memory
-	relayMetaHub := GetRelayMetaHub()
-	relayMeta := relayMetaHub.GetMeta()
-
-	if len(relayMeta.UUID) > 0 {
-		gs, err := mysql.ParseGTIDSet(r.cfg.Flavor, relayMeta.BinlogGTID)
-		if err != nil {
-			return "", terror.ErrRelayLoadMetaData.Delegate(err)
-		}
-		if gs.Contain(gset) {
-			r.tctx.L().Info("get uuid subdir by gtid", zap.Stringer("GTID Set", gset), zap.String("uuid", relayMeta.UUID))
-			return relayMeta.UUID, nil
-		}
-	}
-	return "", terror.ErrNoUUIDDirMatchGTID.Generate(gset.String())
-}
-
-// GetFilePosByGTID tries to get Pos by GTID for file
-func (r *BinlogReader) GetFilePosByGTID(ctx context.Context, filePath string, gset mysql.GTIDSet) (uint32, error) {
+// CheckFileByGTID check whether gset contains file's previous_gset
+func (r *BinlogReader) CheckFileByGTID(ctx context.Context, filePath string, gset mysql.GTIDSet) (bool, error) {
```

Comment:
maybe rename it to `CheckStartSyncFromFile` or `IsGTIDCoverPreviousFiles` or others 😂 current function name shows no usage or behaviour

Reply:
address in 91d1adb